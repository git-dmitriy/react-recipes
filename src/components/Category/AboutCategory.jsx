const AboutCategory = ({ categoryInfo }) => {
  console.log(
    'ðŸš€ ~ file: AboutCategory.jsx ~ line 4 ~ categoryInfo',
    categoryInfo
  );
  const { strCategoryThumb, strCategory, strCategoryDescription } =
    categoryInfo;

  return (
    <div className='hidden sm:block mx-auto xl:max-w-6xl max-w-4xl'>
      <div className='flex bg-yellow-100 rounded-lg p-8 m-2'>
        <div className='sm:w-2/4 w-2/5 bg-yellow-50 rounded-lg flex items-center justify-center'>
          <img className='w-full' src={strCategoryThumb} alt={strCategory} />
        </div>
        <div className='sm:w-2/4 w-3/5 flex flex-col items-start ml-8 pl-8 border-l-2 border-yellow-300'>
          <h4 className='text-xl font-semibold'>{strCategory}</h4>
          <p className='text-md'>{strCategoryDescription}</p>
        </div>
      </div>
    </div>
  );
};

export default AboutCategory;
