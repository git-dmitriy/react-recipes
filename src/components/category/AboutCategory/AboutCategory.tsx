import Lazyload from 'react-lazyload';
import { CategoryItemTypes } from 'appTypes';

type P = {
  categoryInfo: CategoryItemTypes;
};

export const AboutCategory: React.FC<P> = ({ categoryInfo }) => {
  const { strCategory, strCategoryThumb, strCategoryDescription } =
    categoryInfo;

  return (
    <div className=' mx-auto xl:max-w-6xl max-w-4xl'>
      <div className='grid grid-cols-1 md:grid-cols-2 gap-10 p-8 m-2 mb-4'>
        <div className=' flex items-start justify-center'>
          <Lazyload
            once
            offset={100}
            classNamePrefix='w-80 bg-gray-200 rounded-3xl overflow-hidden '
          >
            <img
              className='w-full object-cover'
              src={strCategoryThumb}
              alt={strCategory}
            />
          </Lazyload>
        </div>
        <div className=' flex flex-col justify-around items-start border-gray-400'>
          <h4 className='text-xl font-semibold'>{strCategory}</h4>
          <p className='text-md'>{strCategoryDescription}</p>
        </div>
      </div>
    </div>
  );
};
