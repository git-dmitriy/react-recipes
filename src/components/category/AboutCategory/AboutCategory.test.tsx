import renderer from 'react-test-renderer';
import { AboutCategory } from './AboutCategory';

const data = {
  idCategory: '1',
  strCategory: 'Beef',
  strCategoryDescription:
    'Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]',
  strCategoryThumb: 'https://www.themealdb.com/images/category/beef.png',
};

it('should render correctly', () => {
  const tree = renderer
    .create(<AboutCategory categoryInfo={{ ...data }} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
