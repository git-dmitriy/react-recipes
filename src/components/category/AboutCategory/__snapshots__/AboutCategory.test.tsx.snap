// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div
  className=" mx-auto xl:max-w-6xl max-w-4xl"
>
  <div
    className="grid grid-cols-1 md:grid-cols-2 gap-10 p-8 m-2 mb-4"
  >
    <div
      className=" flex items-start justify-center"
    >
      <div
        className="w-80 bg-gray-200 rounded-3xl overflow-hidden -wrapper "
      >
        <div
          className="w-80 bg-gray-200 rounded-3xl overflow-hidden -placeholder"
          style={
            Object {
              "height": undefined,
            }
          }
        />
      </div>
    </div>
    <div
      className=" flex flex-col justify-around items-start border-gray-400"
    >
      <h4
        className="text-xl font-semibold"
      >
        Beef
      </h4>
      <p
        className="text-md"
      >
        Beef is the culinary name for meat from cattle, particularly skeletal muscle. Humans have been eating beef since prehistoric times.[1] Beef is a source of high-quality protein and essential nutrients.[2]
      </p>
    </div>
  </div>
</div>
`;
